### PR TITLE
Added UI components for MOD-54

### PR DESCRIPTION
**Pull request by enymuss**
Automatically links to required module
Added in the uploadModule screen and showModule screen
Fixed Autoscroll issue [GCI-62]

### CODE REVIEW

File: app/js/showControllers.js [hunk 1]
```diff
@@ -260,6 +273,16 @@ $stateParams, $state) {
 
 })
 
+
+// Link to the required Module
+.filter('linkModule', function($sanitize) {
+  return function(module) {
+    if (!module) return null
+
+      return 'https://modules.openmrs.org/#/show/' + module.slug
```
Comment (elliottwilliams) [inline app/js/showControllers.js]: Use `Config.appUrl` here to pull the URL from the config file (modulusui.conf.js). That way this directive doesn't break on non-production servers.

File: app/partials/createModule.html [hunk 2]
```diff
@@ -94,6 +94,11 @@
       <input class="form-control" type="text" ng-model="release.requiredOMRSVersion">
     </div>
 
+    <div class="form-group">
+      <label>Required Modules</label>
+      <input class="form-control" type="text" ng-model="module.requiredModules" readonly>
```
Comment (enymuss) [inline app/partials/createModule.html]: readonly so it's the same as slug of module

File: app/partials/showModule.html [hunk 3]
```diff
@@ -76,6 +76,17 @@ <h2 ng-if="editable">
       </tr>
 
       <tr>
+
+        <th>Required Modulus</th>
```
Comment (elliottwilliams) [inline app/partials/showModule.html]: Should be "Required Modules" ;)
Modulus = the platform that serves modules
(I make this same typo all the time though)

**Comment by elliottwilliams** [comment]
I'll just merge this and fix that typo :) Thanks so much for this, we've been wanting to build this feature for awhile and it's great to see it finally done!

**Pull request merged**